lib/Dialect/RTG/IR/RTGOps.cpp

```diff
@@ -78,6 +78,78 @@ LogicalResult SetCreateOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// BagCreateOp
+//===----------------------------------------------------------------------===//
+
+ParseResult BagCreateOp::parse(OpAsmParser &parser, OperationState &result) {
+  llvm::SmallVector<OpAsmParser::UnresolvedOperand, 16> elementOperands,
+      multipleOperands;
+  Type elemType;
+
+  if (!parser.parseOptionalLParen()) {
+    while (true) {
+      OpAsmParser::UnresolvedOperand elementOperand, multipleOperand;
+      if (parser.parseOperand(multipleOperand) || parser.parseKeyword("x") ||
+          parser.parseOperand(elementOperand))
+        return failure();
+
+      elementOperands.push_back(elementOperand);
+      multipleOperands.push_back(multipleOperand);
+
+      if (parser.parseOptionalComma()) {
+        if (parser.parseRParen())
+          return failure();
+        break;
+      }
+    }
+  }
+
+  if (parser.parseOptionalAttrDict(result.attributes) || parser.parseColon() ||
+      parser.parseType(elemType))
+    return failure();
+
+  result.addTypes({BagType::get(result.getContext(), elemType)});
+
+  for (auto operand : elementOperands)
+    if (parser.resolveOperand(operand, elemType, result.operands))
+      return failure();
+
+  for (auto operand : multipleOperands)
+    if (parser.resolveOperand(operand, IndexType::get(result.getContext()),
+                              result.operands))
+      return failure();
+
+  return success();
+}
+
+void BagCreateOp::print(OpAsmPrinter &p) {
+  p << " ";
+  if (!getElements().empty())
+    p << "(";
+  llvm::interleaveComma(llvm::zip(getElements(), getMultiples()), p,
+                        [&](auto elAndMultiple) {
+                          auto [el, multiple] = elAndMultiple;
+                          p << multiple << " x " << el;
+                        });
+  if (!getElements().empty())
+    p << ")";
+
+  p.printOptionalAttrDict((*this)->getAttrs());
```

Comment:
nit: Is it intentional that position of optional attribute is different from other operations (bag_selecet and bag_random)?